homeassistant/components/media_player/plex.py

```diff
@@ -272,93 +614,316 @@ def _active_media_plexapi_type(self):
     @property
     def media_content_id(self):
         """Content ID of current playing media."""
-        if self.session is not None:
-            return self._convert_na_to_none(self.session.ratingKey)
+        return self._media_content_id
 
     @property
     def media_content_type(self):
         """Content type of current playing media."""
-        if self.session is None:
-            return None
-        media_type = self.session.type
-        if media_type == 'episode':
+        if self._session_type == 'clip':
+            _LOGGER.debug('Clip content type detected, '
+                          'compatibility may vary: %s',
+                          self.entity_id)
             return MEDIA_TYPE_TVSHOW
-        elif media_type == 'movie':
+        elif self._session_type == 'episode':
+            return MEDIA_TYPE_TVSHOW
+        elif self._session_type == 'movie':
             return MEDIA_TYPE_VIDEO
-        elif media_type == 'track':
+        elif self._session_type == 'track':
             return MEDIA_TYPE_MUSIC
-        return None
+        else:
+            return None
+
+    @property
+    def media_artist(self):
+        """Artist of current playing media, music track only."""
+        return self._media_artist
+
+    @property
+    def media_album_name(self):
+        """Album name of current playing media, music track only."""
+        return self._media_album_name
+
+    @property
+    def media_album_artist(self):
+        """Album artist of current playing media, music track only."""
+        return self._media_album_artist
+
+    @property
+    def media_track(self):
+        """Track number of current playing media, music track only."""
+        return self._media_track
 
     @property
     def media_duration(self):
         """Duration of current playing media in seconds."""
-        if self.session is not None:
-            return self._convert_na_to_none(self.session.duration)
+        return self._media_duration
+
+    @property
+    def media_position(self):
+        """Position of current playing media in seconds.
+
+        Music position appears to update slower than other media.
+        """
+        if self._session:
+            return self._media_position
+
+    @property
+    def media_position_updated_at(self):
+        """When was the position of the current playing media valid."""
+        if self._is_player_active:
+            return util.dt.utcnow()
```

Comment:
"plus update the position when you see a play, pause, stop, next/previous track event" - means I truly grab the position from the client